lib/spack/spack/modules.py

```diff
@@ -79,8 +74,35 @@ def print_help():
             "")
 
 
+def inspect_path(path):
+    class PathInspector(object):
+        dirname2varname = {
+            'bin': ('PATH',),
+            'man': ('MANPATH',),
+            'lib': ('LIBRARY_PATH', 'LD_LIBRARY_PATH'),
+            'lib64': ('LIBRARY_PATH', 'LD_LIBRARY_PATH'),
+            'include': ('CPATH',),
+            'pkgconfig': ('PKG_CONFIG_PATH',)
+        }
+
+        def __call__(self, env, directory, names):
+            for name in names:
+                variables = PathInspector.dirname2varname.get(name, None)
+                if variables is None:
+                    continue
+                absolute_path = join_path(os.path.abspath(directory), name)
+                for variable in variables:
+                    env.prepend_path(variable, absolute_path)
+
+    env, inspector = EnvironmentModifications(), PathInspector()
+    os.path.walk(path, inspector, env)
```

Comment:
Do you really want to walk the whole prefix?  For most of these I think just checking at the top level would be fine.  For pkgconfig, there are a few standard locations you could check (`lib/pkgconfig`, `lib64/pkgconfig`) but I'm not sure you wan to grab all of them unconditionally.  I think for some packages this might pick up stuff you don't want, like tests and such.  I would leave adding the deeper directories to the package implementor --- they can extend it if they want.